src/core/import_species_from_csv.py

```diff
@@ -0,0 +1,170 @@
+# coding: utf8
+
+__contact__ = "info@hytech-imaging.fr"
+__copyright__ = "Copyright (c) 2021 Hytech Imaging"
+
+import csv
+import os.path
+from qgis.core import (
+    QgsVectorLayer,
+    QgsFeature,
+)
+
+
+class ImportSpeciesFromCsv:
+    @staticmethod
+    def importInto(table: QgsVectorLayer):
+        speciesFilePath = os.path.join(
+            os.path.abspath(os.path.dirname(__file__)), "species.csv"
+        )
+        with open(speciesFilePath) as file:
+            lines = file.readlines()
+
+        table.startEditing()
+
+        for i in range(1, len(lines)):
```

Comment:
Something like `for line in lines:` shoud work (instead or retrieveing `lines[i]` later).

Reply:
I used "for i" instead of "for line" because I needed to avoid the first line which contains columns heads.
This is why it is range(1, len ...) instead of range(0, len ...)

Reply:
Slicing is probably more elegant: `for line in lines[1:]:`